src/Discovery/src/Consul/Registry/ConsulRegistration.cs

```diff
@@ -284,7 +284,7 @@ internal static AgentServiceCheck CreateCheck(int port, ConsulDiscoveryOptions o
             check.DeregisterCriticalServiceAfter = DateTimeConversions.ToTimeSpan(options.HealthCheckCriticalTimeout);
         }
 
-        if (options.IsHeartBeatEnabled)
+        if (options.IsHeartBeatEnabled && options.HealthCheckUrl == null && options.HealthCheckPath == ConsulDiscoveryOptions.CONSUL_DEFAULT_HEALTH_PATH)
```

Comment:
@TimHess 
That's the updated line.

Reply:
Thanks for the short summary and explanation @TimHess!
After reading this destilled summary, indeed it doesn't require a fix.

But perhaps adding two unit tests to cover the health check path behavior in relation to heartbeat being on/off could be helpful in the future.

Also one extra test to verify the validation of a negative port for HTTP check.

Thanks for taking your time to look into this and digesting it out in the public 😃